src/components/menu/menu-full-screen/menu-full-screen.component.tsx

```diff
@@ -93,65 +102,98 @@ export const MenuFullscreen = ({
     })
   );
 
+  const closeModal = useCallback(
+    (ev: KeyboardEvent) => {
+      if (onClose && Events.isEscKey(ev)) {
+        ev.stopImmediatePropagation();
+        onClose(ev);
+      }
+    },
+    [onClose]
+  );
+
+  useModalManager({
+    open: isOpen,
+    closeModal,
+    modalRef: menuRef,
+    topModalOverride,
+  });
+
   return (
-    <li aria-label="menu-fullscreen">
+    <li>
       <Portal>
-        <FocusTrap wrapperRef={menuWrapperRef} isOpen={isOpen}>
+        <CSSTransition
+          nodeRef={menuRef}
+          in={isOpen}
+          timeout={transitionDuration}
+          unmountOnExit
+        >
           <StyledMenuFullscreen
-            ref={menuWrapperRef}
-            isOpen={isOpen}
-            menuType={menuType}
+            ref={menuRef}
             startPosition={startPosition}
-            onKeyDown={handleKeyDown}
-            {...rest}
-            {...tagComponent("menu-fullscreen", rest)}
+            transitionDuration={transitionDuration}
           >
-            <StyledMenuFullscreenHeader
-              isOpen={isOpen}
-              menuType={menuType}
-              startPosition={startPosition}
-            >
-              <IconButton
-                aria-label="menu fullscreen close button"
-                onClick={onClose}
-                data-element="close"
-              >
-                <Icon
-                  type="close"
-                  color={isDarkVariant ? "--colorsYang100" : undefined}
-                />
-              </IconButton>
-            </StyledMenuFullscreenHeader>
-            <Box
-              overflowY="auto"
-              scrollVariant={isDarkVariant ? "dark" : "light"}
-              width="100%"
-              height="calc(100% - 40px)"
-            >
-              <StyledMenuWrapper
-                data-component="menu"
+            <FocusTrap wrapperRef={modalRef} isOpen={isOpen}>
+              <StyledMenuModal
+                aria-label={ariaLabel}
+                aria-modal={isTopModal ? true : undefined}
+                data-component="menu-fullscreen"
+                data-element={dataElement}
```

Comment:
comment (non-blocking): if we use the `tagComponent` util here we'd get any updates to supported a `data-` attributes for free as we're already extending `TagProps` above

Reply:
Thanks @edleeks87 👍🏼 Looking at the `tagComponent` and `rootTagTest` utils, I'm not sure these would add significant value, as they seem to abstract some fairly straightforward behaviour to reduce a couple of lines. However, if there are some benefits to using them that I may be overlooking, let me know and I'll add them in

Reply:
The main reason is for DRYness and if we ever change the data- that we support it would automatically be updated here but it's non-blocking